src/compiler/checker.ts

```diff
@@ -27215,6 +27215,7 @@ namespace ts {
                 clear(potentialNewTargetCollisions);
 
                 forEach(node.statements, checkSourceElement);
+                checkSourceElement(node.endOfFileToken);

```

Comment:
Do we need to handle this in `reduceEachChild` and `updateSourceFileNode` as well?

Reply:
Good idea. Do you have any idea of how to test changes to those functions?